CrossAppLogin/Back/src/main/kotlin/com/infomaniak/core/crossapplogin/back/BaseCrossAppLoginViewModel.kt

```diff
@@ -40,18 +50,64 @@ import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.flow.collectLatest
 import kotlinx.coroutines.flow.combine
 import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.flow.transform
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.plus
 import kotlinx.serialization.ExperimentalSerializationApi
+import okhttp3.OkHttpClient
 import com.infomaniak.core.R as RCore
 import com.infomaniak.core.appintegrity.R as RAppIntegrity
 import com.infomaniak.core.network.R as RCoreNetwork
 
+private val apiRepository = object : ApiRepositoryCore() {}
+
+private val baseOkHttpClient by lazy {
+    OkHttpClient.Builder().apply {
+        addCache()
+        addCommonInterceptors()
+    }
+}
+
 abstract class BaseCrossAppLoginViewModel(applicationId: String, clientId: String) : ViewModel() {
+
     private val _availableAccounts = MutableStateFlow(emptyList<ExternalAccount>())
     val availableAccounts: StateFlow<List<ExternalAccount>> = _availableAccounts.asStateFlow()
     val skippedAccountIds = MutableStateFlow(emptySet<Long>())
 
+    val accountsCheckingState: StateFlow<AccountsCheckingState> = _availableAccounts.transform { accounts ->
+        if (accounts.isEmpty()) return@transform
+
+        emit(AccountsCheckingState(status = Ongoing))
+
+        var checkedAccounts = mutableListOf<ExternalAccount>()
+        var checkError: Error = Error.Network
+
+        accounts.forEach { account ->
+            accountCheckStatuses.useElement(account) { statusAsync ->
+                val status = statusAsync.await()
+                when (status) {
+                    Ongoing -> if (checkedAccounts.contains(account).not()) {
+                        checkedAccounts.add(account)
+                        emit(AccountsCheckingState(status = Ongoing, checkedAccounts = { checkedAccounts }))
+                    }
+                    is Error -> if (status is Error.Unknown) checkError = Error.Unknown
+                    WaitingForAccount, Complete -> Unit // Cannot happened, the checkStatuses don't return these values
+                }
+            }
+        }
+
+        emit(
+            AccountsCheckingState(
+                checkedAccounts = { checkedAccounts },
+                status = if (checkedAccounts.isNotEmpty()) Complete else checkError,
+            )
+        )
+    }.stateIn(
+        scope = viewModelScope,
+        started = SharingStarted.Lazily,
```

Comment:
I think we want to make sure we activate this flow as soon as we're the onboarding screen, so it's unlikely the user will need to wait.

Maybe we could launch a collector of this shared flow into `activateUpdates()`, so calling it is all we need to have the thing being warmed up?

Reply:
I don't think a no-op collector would be weird if we add a comment on the why.

Regarding losing some data, there's no problem since it's a `StateFlow`:
The latest data (including the error state, if any) is kept, and we don't need the history, as far as I know.